File: app/src/main/java/com/auto/billiardnote/ui/home/draw/shape/Circle.java
package com.auto.billiardnote.ui.home.draw.shape;

import android.graphics.Canvas;
import android.graphics.Paint;

public class Circle {
    private float x;
    private float y;
    private final float r;
    public Circle(float x, float y, float r)
    {
        this.x = x;
        this.y = y;
        this.r = r;
    }

    public void setCoordinate(float x, float y) {
        this.x = x;
        this.y = y;
    }

    protected boolean isWithin(float x, float y) {
        return (this.x-r) <= x && x <= (this.x+r) && (this.y-r) <= y && y <= (this.y+r);
    }

    public void draw(Canvas canvas, Paint paint) {
        canvas.drawCircle(x, y, r, paint);
    }

    protected float abs(float n, boolean forX) {
        return forX ? Math.abs(n - this.x) : Math.abs(n - this.y);
    }
}

File: app/src/main/java/com/auto/billiardnote/ui/home/draw/shape/Circle.kt
package com.auto.billiardnote.ui.home.draw.shape

import android.graphics.Canvas
import android.graphics.Paint

class Circle {
    private var x: Float
    private var y: Float
    private val r: Float

    constructor(circle: Circle) {
        x = circle.x
        y = circle.y
        r = circle.r
    }

    constructor(x: Float, y: Float, r: Float) {
        this.x = x
        this.y = y
        this.r = r
    }

    fun setCoordinate(x: Float, y: Float) {
        this.x = x
        this.y = y
    }

    fun isWithin(x: Float, y: Float): Boolean {
        return this.x - r <= x && x <= this.x + r && this.y - r <= y && y <= this.y + r
    }

    fun draw(canvas: Canvas, paint: Paint?) {
        canvas.drawCircle(x, y, r, paint!!)
    }

    fun abs(n: Float, forX: Boolean): Float {
        return if (forX) Math.abs(n - x) else Math.abs(n - y)
    }
}